### PR TITLE
Switch from glide to dep.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Since dep requires go 1.8+ I have removed 1.7 from travis.

Thank you for your contribution to Go-AutoRest! We will triage and review it as soon as we can.

As part of submitting, please make sure you can make the following assertions:
 - [ ] I've tested my changes, adding unit tests if applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, except in the case of urgent bug fixes warranting their own release.
 - [ ] If I'm targeting `master`, I've updated [CHANGELOG.md](https://github.com/Azure/go-autorest/blob/master/CHANGELOG.md) to address the changes I'm making.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/go-autorest/issues/225.

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+[[constraint]]
+  branch = "master"
+  name = "github.com/dimchansky/utfbom"
+
+[[constraint]]
+  branch = "master"
+  name = "github.com/mitchellh/go-homedir"
+
+[[constraint]]
+  name = "github.com/stretchr/testify"
```
Comment (mcardosos) [inline Gopkg.toml]: Interesting. I didn't know we were using testify :O

**Comment by mcardosos** [comment]
@jhendrixMSFT  Are you including this in a changelog?

**Comment by jhendrixMSFT** [comment]
@mcardosos I will when we ship v10 (this is in the dev branch as it's a breaking change).

File: Gopkg.toml (hunk 1 above)
Comment (jhendrixMSFT) [reply Gopkg.toml]: Is that bad?  It's been there for quite some time (i.e. was in the glide file).

Comment (mcardosos) [reply Gopkg.toml]: Oh no! It is just surprise. Testify is somewhat popular when doing tests.

**Pull request merged**